src/mscorlib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -289,6 +433,54 @@ public T[] TryPop()
                 Monitor.Exit(this);
                 return arr;
             }
+
+            public void Trim(uint tickCount, int id, MemoryPressure pressure)
+            {
+                if (_count == 0)
+                    return;
+
+                Monitor.Enter(this);
+                uint trimTicks = pressure == MemoryPressure.High ? StackHighTrimAfterTicks : StackTrimAfterTicks;
+                if (_count > 0 && _stockTicks > tickCount || (tickCount - _stockTicks) > trimTicks)
+                {
+                    // We've wrapped the tick count or elapsed enough time since the
+                    // first item went into the stack. Drop the top item so it can
+                    // be collected and make the stack look a little newer.
+
+                    ArrayPoolEventSource log = ArrayPoolEventSource.Log;
+                    int trimCount = StackLowTrimCount;
+                    switch (pressure)
+                    {
+                        case MemoryPressure.High:
+                            trimCount = StackHighTrimCount;
+                            break;
+                        case MemoryPressure.Medium:
+                            trimCount = StackMediumTrimCount;
+                            break;
+                    }
+
+                    while (_count > 0 && trimCount-- > 0)
+                    {
+                        T[] array = _arrays[--_count];
+                        _arrays[_count] = null;
+
+                        if (log.IsEnabled())
+                        {
+                            log.BufferTrimmed(array.GetHashCode(), array.Length, id);
+                        }
+                    }
+
+                    if (_count > 0 && _stockTicks < uint.MaxValue - StackRefreshTicks)
+                        _stockTicks += StackRefreshTicks;
```

Comment:
Style: Since this if isn't directly in the method scope it should have explicit braces.

Reply:
Ok